beacon_node/network/src/network_beacon_processor/mod.rs

```diff
@@ -122,20 +122,22 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
         self: &Arc<Self>,
         message_id: MessageId,
         peer_id: PeerId,
-        aggregate: SignedAggregateAndProof<T::EthSpec>,
+        aggregate: LazySignedAggregateAndProof<T::EthSpec>,
         seen_timestamp: Duration,
     ) -> Result<(), Error<T::EthSpec>> {
         // Define a closure for processing individual attestations.
         let processor = self.clone();
         let process_individual = move |package: GossipAggregatePackage<T::EthSpec>| {
             let reprocess_tx = processor.reprocess_tx.clone();
-            processor.process_gossip_aggregate(
-                package.message_id,
-                package.peer_id,
-                package.aggregate,
-                Some(reprocess_tx),
-                package.seen_timestamp,
-            )
+            if let Ok(signed_aggregate) = package.aggregate.not_lazy() {
+                processor.process_gossip_aggregate(
+                    package.message_id,
+                    package.peer_id,
+                    Box::new(signed_aggregate),
+                    Some(reprocess_tx),
+                    package.seen_timestamp,
+                )
+            }
```

Comment:
if we get an error from `not_lazy` we should return it so it can be handled

Reply:
To deal with error handling should we make the RejectedAggregate type have `LazySignedAggregateAndProof` as its signed_aggregate_field instead `SignedAggregateAndProof` or should we use an enum with Lazy and NotLazy variants?

Reply:
I think adding a variant for Lazy sounds ok. It shouldn't require too many changes because from memory this isn't used in many places